src/resources/lv_img_conv.py

```diff
@@ -0,0 +1,214 @@
+                i = (y*img_width + x)*3 # buffer-index
+                pixel = img.getpixel((x,y))
+                r_act = classify_pixel(pixel[0], 5)
+                g_act = classify_pixel(pixel[1], 6)
+                b_act = classify_pixel(pixel[2], 5)
+                a = pixel[3]
+                if r_act > 0xF8:
+                    r_act = 0xF8
+                if g_act > 0xFC:
+                    g_act = 0xFC
+                if b_act > 0xF8:
+                    b_act = 0xF8
+                c16 = ((r_act) << 8) | ((g_act) << 3) | ((b_act) >> 3) # RGR565
+                buf[i + 0] = (c16 >> 8) & 0xFF
+                buf[i + 1] = c16 & 0xFF
+                buf[i + 2] = a
+                #buf_i0 = buf[i + 0]
+                #buf_i1 = buf[i + 1]
+                #buf_i2 = buf[i + 2]
+                #print(f"{x},{y}: {buf_i0:02x}, {buf_i1:02x}, {buf_i2:02x}")
+                #if (x == 8 and y == 2):
+                #if (x == 10 and y == 0):
+                #    tmp_5 = 1 << (8 - 5)
+                #    tmp_6 = 1 << (8 - 6)
+                #    r = pixel[0]
+                #    g = pixel[1]
+                #    b = pixel[2]
+                #    val_r = round(r / tmp_5) * tmp_5
+                #    val_g = round(g / tmp_6) * tmp_6
+                #    val_b = round(b / tmp_5) * tmp_5
+                #    print("rgba:", r, g, b, a)
+                #    print("rgba_act:", r_act, g_act, b_act, a)
+                #    print("classify_pixel(r, 5):", "tmp:", tmp_5, "r/tmp:", r/tmp_5, "round(r/tmp):", round(r/tmp_5), "val:", val_r)
+                #    print("classify_pixel(g, 6):", "tmp:", tmp_6, "g/tmp:", g/tmp_6, "round(g/tmp):", round(g/tmp_6), "val:", val_g)
+                #    print("classify_pixel(b, 5):", "tmp:", tmp_5, "b/tmp:", b/tmp_5, "round(b/tmp):", round(b/tmp_5), "val:", val_b)
+                #    print("c16:", c16)
+                #    print(f"{x},{y}: {buf_i0:02x}, {buf_i1:02x}, {buf_i2:02x}")
+                #    print("found it!")
+            print(f"done line {y}")
+
+    elif args.color_format == "CF_INDEXED_1_BIT": # ignore binary format, use color format as binary format
+        w = img_width >> 3
+        if img_width & 0x07:
+            w+=1
+        max_p = w * (img_height-1) + ((img_width-1) >> 3) + 8  # +8 for the palette
+        buf = bytearray(max_p+1)
+
+        for y in range(img_height):
+            for x in range(img_width):
+                c, a = img.getpixel((x,y))
+                p = w * y + (x >> 3) + 8  # +8 for the palette
+                #if(!isset(this.d_out[p])) this.d_out[p] = 0  # Clear the bits first
+                buf[p] |= (c & 0x1) << (7 - (x & 0x7))
+        # write palette information
+        palette_size = 2
+        bits_per_value = 1
+        # write 8 palette bytes
+        buf[0] = 0
+        buf[1] = 0
+        buf[2] = 0
+        buf[3] = 0
+        # Normally there is much math behind this, but for the current use case this is close enough
```

Comment:
Could this describe under what scenario this will become a problem (if any)

Reply:
We currently have only one bit per color, and the input images are either transparent, or black. If we had input pictures with different color this information would be lost. I have no idea if this information would be important for a 1-bit-per-pixel image (and I wasn't able to figure out how to do a correct color palette calculation)